lua/spectre_common.lua

```diff
@@ -533,13 +556,15 @@ local function extract_ids_from_string(ids_string)
     return individual_ids, separator
 end
 
-local function fetch_from_cache(id, uri, destination, cache_name, vary_headers)
+local function fetch_from_cache(id, uri, destination, cache_name, vary_headers, custom_module)
     -- Check if datastore already has url cached
     -- Returns the response body. Fills out the the headers
     local start_time = socket.gettime()
-    local cached_value
 
-    cached_value = casper_v2.fetch_body_and_headers(id, uri, destination, cache_name, vary_headers)
+    configure_custom_hooks(custom_module)
+    local cache_key = ngx.ctx.custom_hooks.cache_key(uri)
```

Comment:
I'm afraid `uri` here is not always a uri...
Eg:
```lua
    if request_info.normalized_body ~= nil then
        cache_uri = cache_uri .. tostring(request_info.normalized_body)
    end
```
:)

Reply:
Indeed, although the default hook will just return the `uri` without any changes. If your endpoint `uri` contains something else and you are writing a custom hook, it will need to take this into account.